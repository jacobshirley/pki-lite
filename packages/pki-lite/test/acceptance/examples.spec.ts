import { describe, expect, test } from 'vitest'

const examples = [
    'crl-request',
    'ocsp-request',
    'pfx-bags',
    'self-signed-certificate',
    'timestamp-request',
]

describe('Examples', () => {
    for (const exampleFile of examples) {
        test.sequential(`${exampleFile}`, async () => {
            const logs: string[] = []
            const { log } = console
            console.log = (...args: any[]) => {
                log(...args)
                logs.push(args.map((a) => String(a)).join(' '))
            }

            /* @vite-ignore */
            await import(`../../examples/${exampleFile}.ts`)

            expect(logs).toMatchSnapshot()
            console.log = log
        })
    }
})
