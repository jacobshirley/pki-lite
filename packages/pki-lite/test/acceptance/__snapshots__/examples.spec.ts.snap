// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Examples > crl-request 1`] = `
[
  "CRL Distribution Point Extension: http://localhost:8080/ca.crl",
  "Successfully found CRL [CertificateList] SEQUENCE :
  SEQUENCE :
    INTEGER : 1
    SEQUENCE :
      OBJECT IDENTIFIER : 1.2.840.113549.1.1.11
      NULL
    SEQUENCE :
      SET :
        SEQUENCE :
          OBJECT IDENTIFIER : 2.5.4.6
          PrintableString : 'US'
      SET :
        SEQUENCE :
          OBJECT IDENTIFIER : 2.5.4.8
          UTF8String : 'Test'
      SET :
        SEQUENCE :
          OBJECT IDENTIFIER : 2.5.4.7
          UTF8String : 'Local'
      SET :
        SEQUENCE :
          OBJECT IDENTIFIER : 2.5.4.10
          UTF8String : 'MyOrg'
      SET :
        SEQUENCE :
          OBJECT IDENTIFIER : 2.5.4.11
          UTF8String : 'CA'
      SET :
        SEQUENCE :
          OBJECT IDENTIFIER : 2.5.4.3
          UTF8String : 'MyRootCA'
    UTCTime : 2025-09-21T10:59:16.000Z
    UTCTime : 2025-10-21T10:59:16.000Z
    [0] :
      SEQUENCE :
        SEQUENCE :
          OBJECT IDENTIFIER : 2.5.29.20
          OCTET STRING : 020101
  SEQUENCE :
    OBJECT IDENTIFIER : 1.2.840.113549.1.1.11
    NULL
  BIT STRING : 1000110000001000010100100100100101000110101010001101001100100001100100101000111111000110110111101000110111111010100101011100111001100110111001100001100011100101111110011110101000001110011000011110000110101010100000010101011111111010000101000101101011101100111000010000100111000101001000111000000111110111010011101001100100010100111111111010001110000010011001111110110001001000011010010110100101101010110110010110111000111000010111111100101101000000000000110111001000010111100100101001001100011000110001110010000110111100000111000011110010110100011110111100010000001101110010101001010000001100010101100000011011001011001101001000010111110010101000001001110011110100101000000100111001111000011001000011010100100100110010000000100001011001001111101000100001011010100111011000011010010111101000100100010001001010100011101001000011101011110001001001111100100100111001001111011100111110000100000110001011101001111100010011010110010011010111101111001000100101100100100001110111011011000010110010111000111011111101101110010101111011101001111111110000111111100101100101110001001000101111101100000111110011110110100111010101111011011101000111100101000011110000001011100000011001101111011100111101110101011100110001011010000111100110001000101010110000100101011101001011101001110001010010001110110000000100100010010101000010001100010011110001111000110000001000110011111000110010110101110100000000011101101111011101100010110110101101000111011000000000111001011101001111110010111100111101101001001011110000110011101001111011010001001101001011001100111010110000101111100001110010100001100011101010000010000100010001011110010011110011110000010100100110101010011010111000100110101000000000010001000001011001001011101011010110011110000101111001110100001010000111010001101110111011101110001001000011011101001001010110101111101110101001111010000001000101100111001111111110011011110101001010010111000010000111000001011001010010111011000110101110110010001100111000100100000001011100000100001101000111110000110110001000101110001100100010101111100101010011011101110101001001111010100100101101011001100110110100111101111111110100101001001001010000000011000010100000000100000001101001000100011101110010010011000100010110110010110110001110111100001100111011011010010000111100111111111101010000011000110111100011011100110111111001101001100001101110111101011101111010011101011001000101001001110110110000011110110101110010001001010001010011011001000110001011101010001110111111100111100011110001011011101010010011100101110100000010000001011010101011001010110010001101010100000001010101100000100000101110001001001011111010111110110011001110100100011100111110100111010111001000000101101110101001110110010100111100110101011011101001001101011101001000001101110011011010010010111101010101101001011001001000001001011000110000101010000100011001101110010111000000001010000001101011111011010010101111001000100010110000110010010111010110101000000101110010101101011100000000011001001011000111101010110110011010010011010110000110101000011011110010100110101100011100010001101010110010000011001111110000010101000000010100101101100000100110101110010110000000001000110111010101001011001111011111000110110001110111001001110100100000011100011111101101110101001001001110001001000011111001100001010110000010100011011110111010000010100011110111100111100100100011101111001101101010000010010101000110000101111110011110110111100111001111001000010010001011110110010011010101010111000111110011110110001001111110001000000111011100111011010110111000100010000100101101000110111111101001000110010001011110111100010001000111010110101001000010101100101110011110011010000000110100010110000000100010010000100100001110000000100001111001001100001011011011111110100110111000101101110010100100110110000101100111001000101010111110001000011011100110011000100000001111101100010111010001000101110111010001100101010100010101001010100100101101101101010100010000011011100100011001011000111000011011011001101111111101100000101110101010101111101100011011101001011000110111110000101111000011010100001101001001011001011100100100101101110010000000100001111000110110000010011110",
]
`;

exports[`Examples > ocsp-request 1`] = `
[
  "AIAs: [AuthorityInfoAccess] SEQUENCE :
  [6] :
    IA5String : ''
  [6] :
    IA5String : ''
  [6] :
    IA5String : ''",
  "OCSP Response: [OCSPResponse] SEQUENCE :
  ENUMERATED : 0
  [0] :
    SEQUENCE :
      OBJECT IDENTIFIER : 1.3.6.1.5.5.7.48.1.1
      OCTET STRING : 308206e730820104a1663064310b3009060355040613025553310d300b06035504080c0454657374310e300c06035504070c054c6f63616c310e300c060355040a0c054d794f7267310d300b060355040b0c044f4353503117301506035504030c0e4f43535020526573706f6e646572180f32303235303932313130353931365a30643062304d300906052b0e03021a05000414c3015d4847d542630bdfe6900831b54d0686bdf604142b9ec2875452d6b5e8b2df6b18edf721549811ad021445e08d21404af407b4051ef9dc53e8205c1073dc8200180f32303235303932313130353931365aa1233021301f06092b060105050730010204120410ec9d54aa2c3d29baa2e126a5e0e384a2300d06092a864886f70d01010b0500038201010032c2bafc8e82ce2556a1b9a0d81f03be9633e18ff63252917f91a7068f5139ab728b784afee54da68fac601940fd2a2e0bc5f84745ff64cf573321ba4ab6f505216740933718f92d615bc67fc35db6949bca77aa5963a505688a4fde9218aa17b305b685f3a8c70132cc660e28f0ec6b06da3fa289d26ca0a775af7168cc5a4999c37231754ea5922bde098d22799ba244047b7017fac19d947cfd948d730a2ebd1993c2fd67b5452952baf75242879f0973443634db3c5d552b8463ec84f57fdd0b3edc3ad9d25f02795a6331d361b842f88e1e64ff3df5a5fb8e75362e397b8430ea526617ac7eeb540d8aa4910adfee35ea6eb7050f5177fb44aa52d0904fa08204c7308204c3308204bf308202a7a003020102021445e08d21404af407b4051ef9dc53e8205c1073dd300d06092a864886f70d01010b0500305c310b3009060355040613025553310d300b06035504080c0454657374310e300c06035504070c054c6f63616c310e300c060355040a0c054d794f7267310b3009060355040b0c0243413111300f06035504030c084d79526f6f7443413020170d3235303932313130353931365a180f32303533303230363130353931365a3064310b3009060355040613025553310d300b06035504080c0454657374310e300c06035504070c054c6f63616c310e300c060355040a0c054d794f7267310d300b060355040b0c044f4353503117301506035504030c0e4f43535020526573706f6e64657230820122300d06092a864886f70d01010105000382010f003082010a0282010100bc1edb6e515cdd4184d4240e2835a3a637b70faabe1de2a540be3343c0884d88e166368ab4995388a22312eb3a2454a55323c7d4a452be9f649de9dc4a4b659990ce426c5d96b92716bcb0af4b6b55fd62f3c1d4f2a205c0e9c4af3f2504a3dc59548dd619cbc5470ec608629aec500ac583df95c46376bf1a4f17ba9b5b0fda3fc6b625630628ec2802a36f02325c5c0ef837b4b26a4a0d79f2e51ac862b988157665f257a4434217d6aa36322c0d26bc7d251238764771088eceb8c607f6659145e0c6fd514bf12debce8dde88c706246528d8438c34e2d153b3cdba4a96961cc3b7674d9499992887a5ba866412c95444b885051ad5e7fb5f414269c3e5010203010001a36f306d30090603551d1304023000300b0603551d0f04040302078030130603551d25040c300a06082b06010505070309301d0603551d0e0416041496e7262a6fd98556866a01747e0888388699a3d4301f0603551d230418301680142b9ec2875452d6b5e8b2df6b18edf721549811ad300d06092a864886f70d01010b0500038202010030d3ba2d8daea95881c57ffeba6a7d5991fa977e8df982bdd536094b027222cd389a0b3cc39e2ec60a2547b27612e802cb30579001467281fccaaabc22b5ccff70c31ebf89c1943b98da8bcc661915bc343d19e75a21c35df90d88d508df33005e5563bcc6b954116484314af2eee2442f1d2240d62e06e22a0c59ee3abecba2d79781bb1a36806ac1bdf77d1e0a934b5a194d7d5844fc98a10d5dd2952fe7678ac6fc627667a41f40d518d4bb721974475250272a2f553ee3fe7585dde50aefabeda22db8f3609007de9385afca453ac491999c4c326598f0fc72fa07e4bc314d08bd255750039b28631dbf8ebfb96e74d873b7176cbfbb49078b6965dd5381c666efd69cf210b3eafba151b5e42b38a15cfcdb1702c0e9c343b31f8f15472735a2c1367916973af09204a0d299df5b769b5a716a40319ccc1c9fc5ff325f4760c66743d5449c243b1f5b770b07fa4d9412c06588c2df00183ca56c385d62596dab3f3e350875c836981e9b04fd933980999f8cdc9eb8f5a54c6c1bba3aca210fd23b61d2d5f0cd8ac7ffaf310f1782d3ccb64d72f325c6decb9d92cb8fc9296a8910204d90c20562c64f9df262fb83dc3e091e9d5cdbd4478cdbb4c3d16d6536ef3e969f2db2a5ffad68ca4a82adea0d6cffc610346703805042ef476a2f46ad4c4f40c9c6d9502e7264867febe601257b3bb0f453eef62d8f7865863d24ae",
]
`;

exports[`Examples > pfx-bags 1`] = `
[
  "PFX version: 3",
  "Has MAC data: true",
  "Found 2 bags",
  "Found 1 private keys",
  "Private Key 1: -----BEGIN PRIVATE KEY-----
MIIEowIBAAKCAQEAoQcVFhwtjDUxugBxiq4M0CqBko33IdBZbjR6Jr0EpDs+mzdMGwnhwFRAMF/QZdWIGFWm1o4r5QW11bRlcr7UmLj5n+KbOxNy/Im3dyLrqgCGrZfY2LR9sELkHw/t8AAZF7/v65B5U2ZOWE8nSTpY0tahA64uZJPeo9BOFiSGBGh6eNru1s+Jku9jkTVJpDGNAgDPdEcKO/Cf5eiXsVMxpiEq0ZwIGBzUTsmyFu/u5dqsipbI2L9prkmuY8r8RnwmjZyR4g2XWUPIlYk2tS1EPxPP/ddX9EFttqJ2bZuXWcBJby79z24O2hikfRG+LCPp69Wq1eZYjuUWQ5z69Cl6XwIDAQABAoIBABCv4rQFXIT8mDgdxqZn7gT+eqn800DqTfI5NvVfviTkzNKZ8uWUaxQaxwIcU8TR0YYwDheK6SqVImhQhvjSpMZk2N2ERrDjDZX1EZGjiUd3QX2o0Dx4y9OhwvEjHfWylOccqVjAgootfm3xWfFhPgYvXsU6Xi1nE3nnYsgxpb+8B9tpCL6ahJyZZKdqM8tOuW+meUP4G1rLt9vYwyj28SOzVo6sK7j2wtEQGJVmP9oghGwhN5Fbq3ZxrPB+L37iYQOVRJu76KDc22RyBmAM8ikxT+hGBgshyrmfrflOsq0QyWJt+GnaDI7+DiL0YboaVOxqutXAT4YTJud2J6xs3SkCgYEA1r0NzOow8edlJNUyhhWhIgVLKVJnUdrKieY2O+R9ub1RpwnO6hLz8zosrHHBuT8nAafQ3TWTnQB6PwdSKFpa2FAkHa6wI7Ymb0qbhpqWVnTD0fv9rgPe0yCashZbUYeEowaufly08OWR63ug0Ypms33cHkRLZe4ZzqWgVDsA52cCgYEAv/gAxIgiwkw66HqA9MQqICZr7B0cQ4MqPszhYb9MPSOJ+mdVT26g7LJsX0g7o1iMQaNdw5WhQucGXByrfT3zNaCTpJ5lovO0kXbm3zzSlz0+NzR1EKGELQf27s3GIBkn8BeYzDM/XkKHsqdexItsBLOb6EK0s40cb6Th+kAj0kkCgYAeSssQlhTTex30hTK8YX66it6gdKRokqGxtE0AQ+pMcl17X9HViVjqfdFhfiai2DrLa/31xqINKs0T6Rd4484e56dAnKzyTLa9SlwIL/R3O7+H9rFBhABB7j6EUs0NAj3+tWmWH8PSm7/q5SS5inmUZnPPvTiZp23yjcDBQRWOjwKBgDEYg1HAI2ufxV0Zn4t58/AfA3DyjPzzOcWv7mBqZVwpj6s77ESRXZTLmQ3RUndXOR+8YUmZJhIPNcAvEOjKBD6N39S1Mfc37tFZIHitfIpzfkruHpZvSHef1taCSA6DgIRV9J0c6jL4Pu5BAeqwYaLeMJx5Gu/kwgYHnEe9SawJAoGBAIBbLOGyvtmedooGx1N8nQgYXpwCvLwKdC06QtcU9WrtFvZ/XTD4cUMVTJ4nOnveEQP2KomMYJGguJIhByLl8GTE1vJ58yVWQXXpelJ+ZajW2Gp7xtoRBfn6aCEld25VZ5SpM3LFKHZViqiXVIheNJAeYbnrxTTbV6qnVECW9+q1
-----END PRIVATE KEY-----",
  "Found 1 certificates",
  "Certificate 1: -----BEGIN CERTIFICATE-----
MIID1TCCAr2gAwIBAgIUHXTrpXf+Tb0OOGAIQtrI5DL0RzIwDQYJKoZIhvcNAQELBQAwejELMAkGA1UEBhMCQVUxDTALBgNVBAgMBHRlc3QxDTALBgNVBAcMBHRlc3QxDTALBgNVBAoMBHRlc3QxDTALBgNVBAsMBHRlc3QxETAPBgNVBAMMCHRlc3QuY29tMRwwGgYJKoZIhvcNAQkBFg10ZXN0QHRlc3QuY29tMB4XDTI1MDkyMDEzMTkxMFoXDTI2MDkyMDEzMTkxMFowejELMAkGA1UEBhMCQVUxDTALBgNVBAgMBHRlc3QxDTALBgNVBAcMBHRlc3QxDTALBgNVBAoMBHRlc3QxDTALBgNVBAsMBHRlc3QxETAPBgNVBAMMCHRlc3QuY29tMRwwGgYJKoZIhvcNAQkBFg10ZXN0QHRlc3QuY29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAoQcVFhwtjDUxugBxiq4M0CqBko33IdBZbjR6Jr0EpDs+mzdMGwnhwFRAMF/QZdWIGFWm1o4r5QW11bRlcr7UmLj5n+KbOxNy/Im3dyLrqgCGrZfY2LR9sELkHw/t8AAZF7/v65B5U2ZOWE8nSTpY0tahA64uZJPeo9BOFiSGBGh6eNru1s+Jku9jkTVJpDGNAgDPdEcKO/Cf5eiXsVMxpiEq0ZwIGBzUTsmyFu/u5dqsipbI2L9prkmuY8r8RnwmjZyR4g2XWUPIlYk2tS1EPxPP/ddX9EFttqJ2bZuXWcBJby79z24O2hikfRG+LCPp69Wq1eZYjuUWQ5z69Cl6XwIDAQABo1MwUTAdBgNVHQ4EFgQUY11F+vecBlamihV1DPwuKi1xREYwHwYDVR0jBBgwFoAUY11F+vecBlamihV1DPwuKi1xREYwDwYDVR0TAQH/BAUwAwEB/zANBgkqhkiG9w0BAQsFAAOCAQEAAUETzUusO2mrg76MXwMjY/tKxr7UMq4gxazpGrB015pG+M4Q0PSvry/6Wm7OhPkd6TefnMecTziYl/J/vRk9K7RTkAQa7ABmR6lsEMHmsS4oTSawlMnHCjABFu/4liRhd5WEX+w9bQuKElRCbPUlbfQOouRbJnGBUMe8tqto0IUN7yvOL9AOYBem7zoEVxAqJM7yG/V3pcrF5A3Olm9zhmUrhQIuW60kCP3boqakSBi2+iZ48Q2QjZWHEZx+oWzR97hazyzfkRyQCFV2ERnz7IrXCaGDBCP3csIBDLnR7vX6WG/kDfxPGAC73VG1WegTQYrq3E9fIs2Awb5qa3dUtA==
-----END CERTIFICATE-----",
]
`;

exports[`Examples > self-signed-certificate 1`] = `
[
  "Self-Signed Certificate PEM: -----BEGIN CERTIFICATE-----
MIIDGDCCAgCgAwIBAgIEAQAAADANBgkqhkiG9w0BAQsFADBOMSUwIwYDVQQDExxUZXN0IFNlbGYtU2lnbmVkIENlcnRpZmljYXRlMRgwFgYDVQQKEw9NeSBPcmdhbml6YXRpb24xCzAJBgNVBAYTAlVTMB4XDTIzMDEwMTAwMDAwMFoXDTI0MDEwMTAwMDAwMFowTjElMCMGA1UEAxMcVGVzdCBTZWxmLVNpZ25lZCBDZXJ0aWZpY2F0ZTEYMBYGA1UEChMPTXkgT3JnYW5pemF0aW9uMQswCQYDVQQGEwJVUzCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALfi3KLKMypb+vGmYt77vrPCEtC8lRzm+CPu1N/bVYjyUBDbrmVLC68JYRfu4rlDhkQleohGrPAjkkronCAel588gPtSEIO5+dH/z1ekwKjz8EiSugDUak6Piu+233mOVPJWytRmBof13xwDUyMB4uLmkMsiGoWQ1cF2E9MXSC9qfUL6W3r73rfuN6CUpDNxi32/EJs+oFte0gE760IRJPFPYr2iNlZsF8ZO6snaExe4+7w9svi+DUlzVlSnGHgLUgQxBe+l028mrXUt70FeKnZVJ/VodTDLJ/BEGm5+TFW/Yh6c3TqcdtDYX4aNjRwi68vBibyv7F17YzeBniO7uzECAwEAATANBgkqhkiG9w0BAQsFAAOCAQEApgy4luwOd9T4kl+zmuomnMBINaw7c+DsjoQVpSX4MePKC3lhRwRHI37do+FPmVqk8q5Ul/f2FIU00TU6ogwUaBGSVGsvSRvkUoG/c9Om8KQGKJGwx4TbNGBMy9rhe8yvhYRvjf8+nhEGcydl6+9h09FUSlmxXnfBi+WNNtpBMope+peucevkvUMDJxorl3FZfND+96LgxkGGk5n5YfFsxa5bd1EBcu3Mc4Qp2tozW3BS50MV3x7ihFIQw0P07MNLTqwa8BLehdSaxpUS3WusqRNuqV0Yo7whngmtkjT6xhnCd21NnqKJdkLdi1EQIDpXALBcMFgqRDy7SoZPAoeKaA==
-----END CERTIFICATE-----",
]
`;

exports[`Examples > timestamp-request 1`] = `
[
  "Message hash (SHA-256): dffd6021bb2bd5b0af676290809ec3a53191dd81c7f70a4b28688a362182986f",
  "TimeStampReq created:",
  "- Version: 1",
  "- Hash Algorithm: 2.16.840.1.101.3.4.2.1",
  "- Certificate Requested: true",
  "DER encoded request size: 67 bytes",
  "Response status: granted",
  "Timestamp token received!",
  "Token size: 5461 bytes",
  "Parsed request version: 1",
  "Parsed request certReq: true",
  "TimeStampResp created:",
  "- Status: granted",
  "- Success: true",
  "- Has token: true",
  "DER encoded response size: 5470 bytes",
]
`;
